### PR TITLE
New Check - suspect_pages row limit nearing

### LINKED ISSUE #764: New-Check: msdb.dbo.suspect_pages nearing row limit
## New Check
Number of rows in msdb.dbo.suspect_pages.
 
### What would you like to check?
- any rows in this table indicates that SQL Server has found corrupt pages 
  - depending on the event_type they may no longer be suspect
- if row count gets to 1000, SQL Server stops recording into the table.
  
### What should be configurable for the results of the check ?
Number of rows in the table to alert on?

### What should be able to be excluded from being tested ?
N/A
   
## Additional information    
https://docs.microsoft.com/en-us/sql/relational-databases/backup-restore/manage-the-suspect-pages-table-sql-server?view=sql-server-ver15

### PR DESCRIPTION
**Pull request by jpomfret**
## Please confirm you have 0 failing Pester Tests

[X] There are 0 failing Pester tests
```
Tests completed in 37.36s
Tests Passed: 1575, Failed: 0, Skipped: 0, Pending: 0, Inconclusive: 0 
```

## Changes this PR brings

Please add below the changes that this PR covers. It doesnt need an essay just the bullet points :-)
- Adds a check that closes #764 

Created two config values, one to enable the check to be skipped, another for the % threshold, set at 90% for now, limit is 1000 rows.

Tested by inserting 901 rows into mssql1 suspect pages table
image

### CODE REVIEW

**Comment by jpomfret** [comment]
Oh noes - test failures are below for PSScriptAnalyzer.. worked ok on my machine - is this github actions issue you mentioned?

```
Context Checking PSScriptAnalyzer on Script 'checks/Instance.Tests.ps1'
      [-] The Script Analyzer Rule [PSUseCompatibleCommands] Should not fail 25.22s
        FileNotFoundException: Could not find file 'C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\PSScriptAnalyzer\1.19.0\compatibility_profiles\win-8_x64_10.0.17763.0_6.1.3_x64_4.0.30319.42000_core.json'.
        AggregateException: One or more errors occurred.
        at <ScriptBlock>, D:\a\dbachecks\dbachecks\tests\Project.Tests.ps1: line 39

    Context Checking PSScriptAnalyzer on Script 'internal/configurations/configuration.ps1'
      [-] The Script Analyzer Rule [PSAvoidAssignmentToAutomaticVariable] Should not fail 36ms
        Expected 0, because Script Analyzer says the rules have been broken on lines 5,9 with Message 'The Variable 'input' is an automatic variable that is built into PowerShell, assigning to it might have undesired side effects. If assignment is not by design, please use a different name.' Check in VSCode Problems tab or Run Invoke-ScriptAnalyzer -Script D:\a\dbachecks\dbachecks\internal/configurations/configuration.ps1 -Settings D:\a\dbachecks\dbachecks\PSScriptAnalyzerSettings.psd1, but got 2.
        42:                     $rulefailures.Count | Should -Be 0 -Because "Script Analyzer says the rules have been broken on lines $lines with Message '$message' Check in VSCode Problems tab or Run Invoke-ScriptAnalyzer -Script $ModuleBase\$script -Settings $ModuleBase\PSScriptAnalyzerSettings.psd1"
        at <ScriptBlock>, D:\a\dbachecks\dbachecks\tests\Project.Tests.ps1: line 42
      [-] The Script Analyzer Rule [PSUseCompatibleCommands] Should not fail 25ms
        FileNotFoundException: Could not find file 'C:\Users\runneradmin\Documents\WindowsPowerShell\Modules\PSScriptAnalyzer\1.19.0\compatibility_profiles\win-8_x64_10.0.17763.0_6.1.3_x64_4.0.30319.42000_core.json'.
        AggregateException: One or more errors occurred.
        at <ScriptBlock>, D:\a\dbachecks\dbachecks\tests\Project.Tests.ps1: line 39
```

File: internal/configurations/configuration.ps1 [hunk 1]
```diff
@@ -243,6 +243,7 @@ Set-PSFConfig -Module dbachecks -Name skip.instance.oleautomationproceduresdisab
 Set-PSFConfig -Module dbachecks -Name skip.instance.remoteaccessdisabled -Validation bool -Value $false -Initialize -Description "Skip the remote access check"
 Set-PSFConfig -Module dbachecks -Name skip.instance.scanforstartupproceduresdisabled -Validation bool -Value $false -Initialize -Description "Skip the scan for startup procedures disabled check"
 Set-PSFConfig -Module dbachecks -Name skip.instance.latestbuild -Validation bool -Value $false -Initialize -Description "Skip the scan the latest build of SQL Server check"
+Set-PSFConfig -Module dbachecks -Name skip.instance.suspectpagelimit -Validation bool -Value $false -Initialize -Description "Skip the check for whether the suspect_pages table is nearing the row limit"
```
Comment (ClaudioESSilva) [inline internal/configurations/configuration.ps1:246]: Can we have the number (1000) limit written in the description?

File: internal/configurations/configuration.ps1 [hunk 2]
```diff
@@ -302,6 +303,9 @@ Set-PSFConfig -Module dbachecks -Name command.invokedbccheck.excludedatabases -V
 # config for integration testing
 Set-PSFConfig -Module dbachecks -Name testing.integration.instance -Value @("localhost") -Initialize -Description "Default SQL Server instances to be used by integration tests"
 
+# Suspect pages
+Set-PSFConfig -Module dbachecks -Name policy.suspectpages.threshold -Value 90 -Initialize -Description "Default threshold (%) to check whether suspect_pages is nearing row limit"
```
Comment (ClaudioESSilva) [inline internal/configurations/configuration.ps1:307]: Can we have the number (1000) limit written in the description?

File: internal/configurations/configuration.ps1 (hunk 1 above)
Comment (jpomfret) [reply internal/configurations/configuration.ps1]: Certainly :)

File: internal/configurations/configuration.ps1 (hunk 2 above)
Comment (jpomfret) [reply internal/configurations/configuration.ps1]: Certainly :)

**Comment by jpomfret** [comment]
Noticed this was still out here - @ClaudioESSilva @SQLDBAWithABeard - is there more reviews I haven't fixed? Sorry if I missed one.

**Pull request merged**